keepassxc-browser/content/keepassxc-browser.js

```diff
@@ -719,19 +719,22 @@ kpxc.fillInFromActiveElement = async function(passOnly = false) {
     }
 
     const el = document.activeElement;
-    if (el.nodeName !== 'INPUT') {
-        // No active input element selected -> fill the first combination found
-        if (kpxc.combinations.length > 0) {
-            kpxc.fillInCredentials(kpxc.combinations[0], kpxc.credentials[0].login, kpxc.credentials[0].uuid, passOnly);
-
-            // Focus to the input field
-            const field = passOnly ? kpxc.combinations[0].password : kpxc.combinations[0].username;
-            if (field) {
-                field.focus();
-            }
+    if (el.nodeName !== 'INPUT' && kpxc.combinations.length > 0 && kpxc.settings.autoCompleteUsernames) {
```

Comment:
Shouldn't we be checking if the active element is not the first kpxc combination field? We could theoretically be focused on a search box input field (for example) and result in the bug again.

Reply:
My thought on this in pseudo code would be:

```
if (autocomplete_enabled)
	focus on kpxc.combinations[0]
	if (credentials > 1)
		show choice list
	else
		fill credentials
	return
endif
```

Reply:
@droidmonkey this would simplify the code, I guess.

But the check for `kpxc.combinations.length > 0` would still be required, right? So the initial `if` would be:
`if (kpxc.combinations.length > 0 && kpxc.settings.autoCompleteUsernames) {`?

And that would enclose current lines 722 - 741.

Is the part from line 743 on the only remaining case then?

Reply:
Good point, yes need to ensure credentials are even available. I would put the autocomplete conditional first since that is the main purpose of the block.

Reply:
This should just be:
```
    if (el.nodeName !== 'INPUT') {
        // No active input element selected -> fill the first combination found
        if (kpxc.combinations.length > 0) {
            // Focus to the input field
            const field = passOnly ? kpxc.combinations[0].password : kpxc.combinations[0].username;
            if (field) {
                field.focus();
            }

            if (kpxc.credentials.length === 1) {
                kpxc.fillInCredentials(kpxc.combinations[0], kpxc.credentials[0].login, kpxc.credentials[0].uuid, passOnly);
                return;
            }

            kpxcAutocomplete.showList(field);
        }

        return;
    }
```

Reply:
Why does it matter if the current element is not an input field? Also, we only want to do this if auto complete is active.

Reply:
@droidmonkey Because it indicates we are not focused to any input field when trying to do the fill from keyboard. But true, the check for AutoComplete option is also needed.

Reply:
@varjolintu shouldn't `fillInCredentials` and `showList' be inside the `if (field)` block?

Reply:
Yes.

Reply:
I think @droidmonkey was right. It is not sufficient to check for `nodeName !== 'INPUT'`. In that case the autocomplete list would open in any input field that is currently focused, if `kpxc.credentials.length > 1`.

For example go to https://www.otto.de/user/login and click into the search field at the top. Then press `Alt-Shift-u`. If you have multiple logins the list would open beneath the search field.